src/main/java/seedu/uninurse/logic/commands/EditMedicationCommand.java

```diff
@@ -0,0 +1,173 @@
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: PATIENT_INDEX (must be a positive integer) "
+            + "MEDICATION_INDEX (must be a positive integer) "
+            + PREFIX_MEDICATION + "MEDICATION_TYPE | MEDICATION_DOSAGE\n"
+            + "Example: " + COMMAND_WORD + " 2 " + " 1 "
+            + PREFIX_MEDICATION + "Ampicillin | 0.5 IV every 6 hours";
+
+    public static final String MESSAGE_EDIT_MEDICATION_SUCCESS = "Edited medication %1$d of %2$s:\n"
+            + "Before: %3$s\n"
+            + "After: %4$s";
+    public static final String MESSAGE_NOT_EDITED = "Medication has not been modified.";
+    public static final String MESSAGE_EDIT_DUPLICATE_MEDICATION =
+            "Medication already exists in %1$s's medication list.";
+
+    public static final CommandType EDIT_MEDICATION_COMMAND_TYPE = CommandType.EDIT_PATIENT;
+
+    private final Index patientIndex;
+    private final Index medicationIndex;
+    private final EditMedicationDescriptor editMedicationDescriptor;
+
+    /**
+     * Creates an EditMedicationCommand to edit a {@code Medication} from the specified patient.
+     *
+     * @param patientIndex of the patient in the filtered patient list to edit.
+     * @param medicationIndex of the medication to be edited.
+     * @param editMedicationDescriptor descriptor with details to edit for the medication.
+     */
+    public EditMedicationCommand(Index patientIndex, Index medicationIndex,
+            EditMedicationDescriptor editMedicationDescriptor) {
+        requireAllNonNull(patientIndex, medicationIndex, editMedicationDescriptor);
+
+        this.patientIndex = patientIndex;
+        this.medicationIndex = medicationIndex;
+        this.editMedicationDescriptor = editMedicationDescriptor;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Patient> lastShownList = model.getFilteredPersonList();
+
+        if (patientIndex.getZeroBased() >= lastShownList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+        }
+
+        Patient patientToEdit = lastShownList.get(patientIndex.getZeroBased());
+        MedicationList initialMedicationList = patientToEdit.getMedications();
+
+        if (medicationIndex.getZeroBased() >= initialMedicationList.size()) {
+            throw new CommandException(Messages.MESSAGE_INVALID_MEDICATION_INDEX);
+        }
+
+        Medication initialMedication = initialMedicationList.get(medicationIndex.getZeroBased());
+        Medication updatedMedication = new Medication(
+                editMedicationDescriptor.getType().orElse(initialMedication.getType()),
+                editMedicationDescriptor.getDosage().orElse(initialMedication.getDosage()));
+
+        if (initialMedicationList.getInternalList().contains(updatedMedication)) {
```

Comment:
Would having a `MedicalCondition#contains()` method be better, instead of using the internal list?
```suggestion
        if (initialMedicationList.contains(updatedMedication)) {
```